mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/NativeQueryParser.java

```diff
@@ -105,6 +104,8 @@ public Rule Operation() {
 				Sequence( FindOne(), builder.setOperation( Operation.FINDONE ) ),
 				Sequence( FindAndModify(), builder.setOperation( Operation.FINDANDMODIFY ) ),
 				Sequence( Insert(), builder.setOperation( Operation.INSERT ) ),
+				Sequence( InsertOne(), builder.setOperation( Operation.INSERTONE ) ),
+				Sequence( InsertMany(), builder.setOperation( Operation.INSERTMANY ) ),
 				Sequence( Remove(), builder.setOperation( Operation.REMOVE ) ),
```

Comment:
> Breafly, MongoCollection has not method "insert", but mongodb shell has. In this way, I must to support all methods "insert" in the PR

You can certainly do that in a separate PR like I asked you. In reality, you are adding a new feature to our parser that consist in understanding the inertOne and insertMany construct. It would deserve at least to be included in a separate commit. The reason I'm saying this is a new feature is because you didn't remove the old doInsert method. In the end you are not replacing the deprecated API.

Also, I'm not sure what's the criteria for adding insertOne and insertMany and not the new update and delete methods.